.github/workflows/release.yml

```diff
@@ -41,6 +40,32 @@ jobs:
           python setup.py sdist bdist_wheel
           twine upload dist/*
 
+  publish-merlin-pyfunc-server:
+    # if: ${{ startsWith(github.ref, 'refs/tags/') }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Comment:
Setting fetch-depth to 0 so it can download all commits history for tags. If not set, the $tag value from the setuptools_scm command would be wrong (i.e.: 0.1.dev1 here: https://github.com/caraml-dev/merlin/actions/runs/8259794693/job/22594492670#step:5:382)